packages/fx-core/src/plugins/resource/bot/languageStrategy.ts

```diff
@@ -101,7 +96,7 @@ export class LanguageStrategy {
     if (programmingLanguage === ProgrammingLanguage.JavaScript) {
       try {
         // fail to npm install @microsoft/teamsfx on azure web app, so pack it locally.
-        await utils.execute("npm install", packDir);
+        await utils.execute(`${Commands.NPM_INSTALL}`, packDir);
```

Comment:
Since we put build from remote to local, have we handled the incremental build/deployment case in the current code? e.g. if there is no new file change, there is no need to build the artifact again and we don't need to deploy the same artifact to Azure. This is another performance improvement of resource plugin, you can find more details with Heng and check if it apply to Bot plugin

Reply:
Beside, since we have changed the build and deployment strategy/solution, it is a good opportunity for us to review the current Bot design on provision/deployment again to check if there is any other place need to be updated along with this design change?